### PR TITLE
[Mutants] Improve MixedDataSet

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

**Review by vjik — COMMENTED** [review]
Why need exception?

File: src/Validator.php [hunk 1]
```diff
@@ -91,7 +92,12 @@ public function validate(mixed $data, iterable|object|string|null $rules = null)
             if (is_int($attribute)) {
                 $validatedData = $data->getData();
             } else {
-                $validatedData = $data->getAttributeValue($attribute);
+                try {
+                    $validatedData = $data->getAttributeValue($attribute);
+                } catch (BadMethodCallException) {
+                    throw new InvalidArgumentException('Properties can\'t be used in rules with single value data set');
```
Comment (samdark) [inline src/Validator.php:98]:
```suggestion
                    throw new InvalidArgumentException('Properties can\'t be used in rules with single value data set.');
```

File: tests/ValidatorTest.php [hunk 2]
```diff
@@ -86,9 +87,23 @@ public function testDiverseTypes($dataSet): void
     public function testNullAsDataSet(): void
     {
         $validator = ValidatorFactory::make();
-        $result = $validator->validate(null, ['property' => [new CompareTo(null)]]);
+        $result = $validator->validate(null, [[new CompareTo(null)]]);
+        $this->assertTrue($result->isValid());
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Properties can\'t be used in rules with single value data set');
```
Comment (samdark) [inline tests/ValidatorTest.php:94]:
```suggestion
        $this->expectExceptionMessage('Properties can\'t be used in rules with single value data set.');
```

File: tests/ValidatorTest.php [hunk 3]
```diff
@@ -86,9 +87,23 @@ public function testDiverseTypes($dataSet): void
...
+    }
 
+    public function testSingleValueAsDataSet(): void
+    {
+        $validator = ValidatorFactory::make();
+        $result = $validator->validate(1, [[new InRange([1, 2])]]);
         $this->assertTrue($result->isValid());
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Properties can\'t be used in rules with single value data set');
```
Comment (samdark) [inline tests/ValidatorTest.php:105]:
```suggestion
        $this->expectExceptionMessage('Properties can\'t be used in rules with single value data set.');
```

**Review by samdark — APPROVED** [review]

**Comment by arogachev** [comment]
The changes that are not ambiguous extracted to #354.

**Comment by arogachev** [comment]
Renaming part was extracted and merged separately. As for the main idea, it was rejected by other team members, even after suggesting an alternative - splitting interfaces. Closing, not worth syncing / creating a separate issue.

**Pull request closed without merging**